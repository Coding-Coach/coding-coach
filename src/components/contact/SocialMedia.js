import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faFacebook, faTwitter, faInstagram } from '@fortawesome/free-brands-svg-icons';

import styles from './assets/socialmedia.scss';

const SocialMedia = () => {
  return (
    <div className={styles.social_media}>
      <div className={styles.icon_wrapper}>
        <a target="_blank" href={process.env.REACT_APP_FB_URL} rel="noreferrer noopener">
          <FontAwesomeIcon icon={faFacebook} size="2x" className={styles.green_icons} />
        </a>
      </div>
      <div className={styles.icon_wrapper}>
        <a target="_blank" href={process.env.REACT_APP_INSTA_URL} rel="noreferrer noopener">
          <FontAwesomeIcon icon={faInstagram} size="2x" className={styles.green_icons} />
        </a>
      </div>
      <div className={styles.icon_wrapper}>
        <a target="_blank" href={process.env.REACT_APP_TWITTER_URL} rel="noreferrer noopener">
          <FontAwesomeIcon icon={faTwitter} size="2x" className={styles.green_icons} />
        </a>
      </div>
    </div>
  );
};

export default SocialMedia;
