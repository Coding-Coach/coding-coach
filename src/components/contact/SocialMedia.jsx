import React from 'react';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faFacebook, faTwitter, faInstagram } from "@fortawesome/free-brands-svg-icons";

import styles from './assets/socialmedia.scss';

const SocialMedia = () => {
  return (
    <div className={styles.social_media}>
      <div className={styles.icon_wrapper}>
        <a target="_blank" href={process.env.REACT_APP_FB_URL}>
          <FontAwesomeIcon icon={faFacebook} size='2x' className={styles.green_icons} />
        </a>
      </div>
      <div className={styles.icon_wrapper}>
        <a target="_blank" href={process.env.REACT_APP_INSTA_URL}>
          <FontAwesomeIcon icon={faInstagram} size='2x' className={styles.green_icons} />
        </a>
      </div>
      <div className={styles.icon_wrapper}>
        <a target="_blank" href={process.env.REACT_APP_TWITTER_URL}>
          <FontAwesomeIcon icon={faTwitter} size='2x' className={styles.green_icons} />
        </a>
      </div>
    </div>
  );
};

export default SocialMedia;
