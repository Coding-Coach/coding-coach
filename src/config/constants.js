export default {
  contact: {
    EMAIL: process.env.REACT_APP_EMAIL || 'codingcoachio@gmail.com',
    SLACK_URL:
      process.env.REACT_APP_SLACK_URL ||
      'https://join.slack.com/t/coding-coach/shared_invite/enQtNDMyMzUzNjAxODQyLTUwYTFkMzZmNGFhMzZjYTQwOWE2YWFjOGJhNzYyOGIxNDM5Zjc1YWQ4NjMwN2U5YzBlNjkwMTI2ZDNiOWQyMzM',
  },
  social: {
    FB_URL: process.env.REACT_APP_FACEBOOK_URL || 'https://www.facebook.com/codingcoachio/',
    INSTA_URL: process.env.REACT_APP_INSTA_URL || 'https://www.instagram.com/codingcoach_io/',
    TWITTER_URL: process.env.REACT_APP_TWITTER_URL || 'https://twitter.com/codingcoach_io',
  },
  legal: {
    PRIVACY_POLICY_URL:
      process.env.REACT_APP_PRIVACY_POLICY_URL ||
      'https://www.freeprivacypolicy.com/free-privacy-policy-generator.php',
  },
  payments: {
    paypal: process.env.REACT_APP_PAYPAL_DONATE_BUTTON_ID || '123abc',
  },
};
