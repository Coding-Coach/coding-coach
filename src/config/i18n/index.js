import en from './en/messages.js';
import es from './es/messages.js';
import fr from './fr/messages.js';

export const catalogs = { en, es, fr };
