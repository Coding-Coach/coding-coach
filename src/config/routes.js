const routesConfiguration = {
  landing: {
    name: 'landing_home',
    home: '/',
  },
  dashboard: {
    name: 'home',
    group: 'dashboard',
    path: '/app/dashboard',
  },
  messages: {
    name: 'messages',
    group: 'messages',
    path: '/app/messages',
  },
  mentorships: {
    name: 'mentorships',
    group: 'mentorships',
    path: '/app/mentorships',
  },
  settings: {
    name: 'settings',
    group: 'settings',
    path: '/app/settings',
  },
};

export { routesConfiguration };
