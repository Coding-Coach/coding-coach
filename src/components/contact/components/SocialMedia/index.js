import SocialMedia from './SocialMedia';

export default SocialMedia;
