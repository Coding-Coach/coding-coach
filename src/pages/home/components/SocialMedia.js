import React from 'react';
import SocialIcon from 'components/socialIcon/SocialIcon';
import { faFacebook, faTwitter, faGithub } from '@fortawesome/free-brands-svg-icons';

import config from 'config/constants';

const SocialMedia = () => {
  return (
    <div className="flex flex-row-reverse self-center md:self-auto">
      <SocialIcon href={config.social.GITHUB_URL} icon={faGithub} size="2x" className="mr-0" />
      <SocialIcon href={config.social.FB_URL} icon={faFacebook} size="2x" />
      <SocialIcon href={config.social.TWITTER_URL} icon={faTwitter} size="2x" className="ml-0" />
    </div>
  );
};

export default SocialMedia;
