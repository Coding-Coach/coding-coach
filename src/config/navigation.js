import { routesConfiguration } from 'config/routes';
import IconDashboard from 'components/icon/Dashboard';
import IconMessages from 'components/icon/Messages';
import IconMentorships from 'components/icon/Mentorships';
import IconSettings from 'components/icon/Settings';

const navConfiguration = {
  tabs: [
    {
      name: routesConfiguration.dashboard.name,
      path: routesConfiguration.dashboard.path,
      group: routesConfiguration.dashboard.group,
      label: 'dashboard.navigation.tab.home',
      icon: IconDashboard,
    },
    {
      name: routesConfiguration.messages.name,
      path: routesConfiguration.messages.path,
      group: routesConfiguration.messages.group,
      label: 'dashboard.navigation.tab.messages',
      icon: IconMessages,
    },
    {
      name: routesConfiguration.mentorships.name,
      path: routesConfiguration.mentorships.path,
      group: routesConfiguration.mentorships.group,
      label: 'dashboard.navigation.tab.mentorships',
      icon: IconMentorships,
    },
    {
      name: routesConfiguration.settings.name,
      path: routesConfiguration.settings.path,
      group: routesConfiguration.settings.group,
      label: 'dashboard.navigation.tab.settings',
      icon: IconSettings,
    },
  ],
};

export { navConfiguration };
