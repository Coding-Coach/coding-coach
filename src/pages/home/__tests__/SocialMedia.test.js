import React from 'react';
import { render } from 'react-testing-library';
import SocialMedia from '../components/SocialMedia';
import config from 'config/constants';

test('renders the social media icons', () => {
  const { container } = render(<SocialMedia />);

  expect(container.firstChild).toMatchSnapshot();

  const href = container.querySelectorAll('a');
  expect(href[0].getAttribute('href')).toEqual(config.social.FB_URL);
  expect(href[1].getAttribute('href')).toEqual(config.social.INSTA_URL);
  expect(href[2].getAttribute('href')).toEqual(config.social.TWITTER_URL);
});
