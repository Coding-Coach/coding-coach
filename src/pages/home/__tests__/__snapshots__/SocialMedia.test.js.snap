// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the social media icons 1`] = `
<div>
  <div>
    <a
      target="_blank"
      href="https://www.facebook.com/codingcoachio/"
      rel="noreferrer noopener"
    >
      <svg
        aria-hidden="true"
        data-prefix="fab"
        data-icon="facebook"
        class="svg-inline--fa fa-facebook fa-w-14 fa-2x "
        role="img"
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 448 512"
      >
        <path
          fill="currentColor"
          d="M448 56.7v398.5c0 13.7-11.1 24.7-24.7 24.7H309.1V306.5h58.2l8.7-67.6h-67v-43.2c0-19.6 5.4-32.9 33.5-32.9h35.8v-60.5c-6.2-.8-27.4-2.7-52.2-2.7-51.6 0-87 31.5-87 89.4v49.9h-58.4v67.6h58.4V480H24.7C11.1 480 0 468.9 0 455.3V56.7C0 43.1 11.1 32 24.7 32h398.5c13.7 0 24.8 11.1 24.8 24.7z"
        />
      </svg>
    </a>
  </div>
  <div>
    <a
      target="_blank"
      href="https://www.instagram.com/codingcoach_io/"
      rel="noreferrer noopener"
    >
      <svg
        aria-hidden="true"
        data-prefix="fab"
        data-icon="instagram"
        class="svg-inline--fa fa-instagram fa-w-14 fa-2x "
        role="img"
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 448 512"
      >
        <path
          fill="currentColor"
          d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
        />
      </svg>
    </a>
  </div>
  <div>
    <a
      target="_blank"
      href="https://twitter.com/codingcoach_io"
      rel="noreferrer noopener"
    >
      <svg
        aria-hidden="true"
        data-prefix="fab"
        data-icon="twitter"
        class="svg-inline--fa fa-twitter fa-w-16 fa-2x "
        role="img"
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 512 512"
      >
        <path
          fill="currentColor"
          d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
        />
      </svg>
    </a>
  </div>
</div>
`;
